### ISSUE TITLE
Bug: Rule Editor breaks

### ISSUE THREAD

**Issue by thomashempel** [deepnight/led#107]
Hi,

the rule editor for auto tiles breaks for me. When I add more than two rules.

The first two rules are displayed in the list and can be edited. The third one is displayed weird and can't be edited. The rule itself applies, though.

![Screenshot 2020-09-27 122656](https://user-images.githubusercontent.com/354084/94362684-6cda5900-00bd-11eb-947a-a8cf59af3222.png)

The small "+" icons don't do anything but one can add a new rule with the "+" icon in the group. After adding the rule, only one broken rule is displayed which can't be edited. Again, the rules themself are applied correctly.

![Screenshot 2020-09-27 122748](https://user-images.githubusercontent.com/354084/94362714-9b583400-00bd-11eb-9634-f8e63a00525f.png)

When I look in to the JSON, there a lot of empty rules, which I did not create. Even if I delete them, the problem persists.

Here is the AutoLayer section from the project JSON:

```json
{
  "identifier": "AutoLayer",
  "type": "AutoLayer",
  "uid": 14,
  "gridSize": 16,
  "displayOpacity": 1.0,
  "intGridValues": [ { "identifier": null, "color": "#000000" } ],
  "autoTilesetDefUid": 1,
  "autoRuleGroups": [ {
    "uid": 26,
    "name": "walls",
    "active": true,
    "collapsed": false,
    "rules": [
      {
        "uid": 44,
        "active": true,
        "size": 3,
        "tileIds": [ 4 ],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 2, 2, 0, 1, 2, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 5013495,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 43,
        "active": true,
        "size": 3,
        "tileIds": [ 8 ],
        "chance": 1.0,
        "pattern": [ 0, 2, 0, 1, 2, 0, 0, 2, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 8750498,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 27,
        "active": true,
        "size": 3,
        "tileIds": [ 1 ],
        "chance": 1.0,
        "pattern": [ 0, 1, 0, 0, 2, 0, 0, 0, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 7641868,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 28,
        "active": true,
        "size": 3,
        "tileIds": [ 17 ],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 0, 2, 0, 0, 1, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 2162530,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 36,
        "active": true,
        "size": 3,
        "tileIds": [ 10 ],
        "chance": 1.0,
        "pattern": [ 0, 2, 0, 0, 2, 1, 0, 2, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 7500645,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 42,
        "active": true,
        "size": 3,
        "tileIds": [],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 9008499,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 41,
        "active": true,
        "size": 3,
        "tileIds": [],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 440326,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 40,
        "active": true,
        "size": 3,
        "tileIds": [],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 561342,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 39,
        "active": true,
        "size": 3,
        "tileIds": [],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 4198554,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 38,
        "active": true,
        "size": 3,
        "tileIds": [],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 8545260,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      },
      {
        "uid": 37,
        "active": true,
        "size": 3,
        "tileIds": [],
        "chance": 1.0,
        "pattern": [ 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
        "flipX": false,
        "flipY": false,
        "perlinActive": false,
        "perlinSeed": 6478714,
        "perlinScale": 0.2,
        "perlinOctaves": 2
      }
    ]
  } ],
  "autoSourceLayerDefUid": 6,
  "tilesetDefUid": null,
  "tilePivotX": 0,
  "tilePivotY": 0
}
```

BTW. Thank you for this tool. I decided to not build my own editor in-game and instead use this. ;-)

**Comment by deepnight** [comment]
I think it's related to a crash I fixed in 0264c304f0045cd31e5f2814c0e947662b555084 . It happens only with "pure" auto-layers. It will be fixed in today release :)

**Comment by thomashempel** [comment]
This is fixed now with 0.3! thanks
